telegram/_utils/repr.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2023
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains auxiliary functionality for building strings for __repr__ method.
+
+Warning:
+    Contents of this module are intended to be used internally by the library and *not* by the
+    user. Changes to this module are not considered breaking changes and may not be documented in
+    the changelog.
+"""
+from typing import Any
+
+
+def build_repr_with_selected_attrs(obj: object, **kwargs: Any) -> str:
+    """Create ``__repr__`` string in the style `Classname(arg1=1, arg2=2).`
+
+    Attributes that are to be used in the representation, are passed as kwargs.
+    """
+
+    def stringify(key: str, val: Any) -> str:
+        return f"{key}={val.__qualname__ if callable(val) else val}"
```

Comment:
this could be moved outside of `build_repr_*` and made protected with a leading underscore.

Reply:
Done. It's an interesting, almost holywar-like discussion about functions inside of functions. I've heard many arguments pro and contra over these months, what's yours, in a nutshell, for not having one function inside another?

Reply:
I would expect that there is a (probably tiny) overhead in defining the inner function on every call to the outer function. But I have to "proper" argumentation at hand for my suggestion :D